### PR TITLE
Only change expiration date if it is different

### LINKED ISSUE #13235: User module - `expires` option always fires changed
##### Issue Type:

Bug Report

##### Component Name:
user

##### Ansible Version:

1.9.4
##### Ansible Configuration:

N/A, no significant changes.
##### Environment:

CentOS 7
##### Summary:

The `user` module will always fire 'changed' when the `expires` option is set. 
##### Steps To Reproduce:

Use the `expires` option when creating a user. 
##### Expected Results:

Run the playbook for the first time. The user is created with an expiration date, and 'changed' is fired. 

Run the playbook a second time. Nothing has changed, and 'changed' is not fired. 
##### Actual Results:

Run the playbook for the first time. The user is created with an expiration date, and 'changed' is fired. 

Run the playbook a second time. Nothing has changed, and 'changed' is fired.

### PR DESCRIPTION
**Pull request by samdoran**
##### SUMMARY

Modify `user_info()` method to also return the password expiration.
Compare current and desired expiration times and only change if they are different.

Fixes #13235

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME
`user.py`

##### ANSIBLE VERSION

```
2.5
```

### CODE REVIEW

**Comment by webknjaz** [comment]
-label needs_triage

**Comment by bcoca** [comment]
@samdoran, i believe the failing tests are wrong

**Comment by samdoran** [comment]
@bcoca Looks like some tests are incorrect, others are legit.

I need to skip for macOS. The [failure](https://app.shippable.com/github/ansible/ansible/runs/61682/28/tests) on CentOS 6 with Python 2.6 is legit: no `totalseconds()` method ☹️. Guess I'll have to do that differently.

Also, I need to add expiration change logic for FreeBSD as well. I missed that since I only fixed the method in the User class.

Some of the tests, I'm not sure what is triggering the change on the second task run. [RHEL7.4/1](https://app.shippable.com/github/ansible/ansible/runs/61682/25/tests) for example.

File: lib/ansible/modules/system/user.py [hunk 1]
```diff
@@ -271,6 +270,7 @@ class User(object):
     platform = 'Generic'
     distribution = None
     SHADOWFILE = '/etc/shadow'
+    SHADOWFILE_INDEX = 7
```
Comment (bcoca) [inline lib/ansible/modules/system/user.py]: shadowfile_expire_index? .. just index seems ambiguous and we might need others in future

Comment (samdoran) [reply lib/ansible/modules/system/user.py]: Very good point.

File: test/integration/targets/user/tasks/main.yml [hunk 2]
```diff
@@ -20,142 +20,200 @@
...
-          - "user_test1.results[0]['state'] == 'present'"
-          - "user_test1.results[1]['state'] == 'present'"
-          - "user_test1.results[2]['state'] == 'present'"
-          - "user_test1.results[3]['state'] == 'present'"
-          - "user_test1.results[4]['state'] == 'present'"
-  when: "jinja2_version.stdout is version('2.6', '<')"
-
-##
+    that:
+      - "not user_test1.results[0]['changed']"
```
Comment (bcoca) [inline test/integration/targets/user/tasks/main.yml]: the proper test would be ` is changed`

**Comment by samdoran** [comment]
@bcoca So the failing tests revealed a couple problems:

1. I used the incorrect date math functions that mix localtime and UTC.
2. The `timezone` integration tests change the timezone on the system and don't set it back to UTC.

So the untidy `timezone` test actually exposed a bug in my fix. Lucky for me.

I'm working on getting this fixed now that I've been able to figure out what's going on (with a ton of help from @mattclay).

**Comment by mattclay** [comment]
CI failure in FreeBSD integration tests: 

https://app.shippable.com/github/ansible/ansible/runs/62723/26/tests
https://app.shippable.com/github/ansible/ansible/runs/62723/27/tests

The other failures are unrelated to this PR.

**Comment by samdoran** [comment]
@bcoca I think I finally got this working.

I found a bug in the FreeBSD class. Passing an int to `-e` sets that directly in `/etc/master.passwd`:

```
root@ip-192-168-3-46:~/ansible$ grep ansi /etc/master.passwd
ansibulluser:*:1002:1002::0:6221:User &:/home/ansibulluser:/bin/sh
                            ^
                            This is seconds since epoch
```

The man pages aren't clear and some of them do say an integer in DAYS or a date string. I switched to using the date string since that seems to behave correctly.

**Pull request merged**